src/contracts/core/PaymentCoordinator.sol

```diff
@@ -110,88 +106,328 @@ contract PaymentCoordinator is
+            require(
+                block.timestamp - LOWER_BOUND_START_RANGE <= rangePayment.startTimestamp,
+                "PaymentCoordinator._payForRange: startTimestamp too far in the past"
+            );
+        } else {
+            require(
+                block.timestamp <= rangePayment.startTimestamp,
+                "PaymentCoordinator._payForRange: startTimestamp cannot be in the past"
+            );
+        }
         require(
-            (retroactivePaymentsEnabled && rangePayment.startTimestamp >= LOWER_BOUND_START_RANGE) ||
-                rangePayment.startTimestamp >= block.timestamp,
-            "PaymentCoordinator.payForRange: invalid startTimestamp set"
+            rangePayment.startTimestamp <= block.timestamp + UPPER_BOUND_START_RANGE,
+            "PaymentCoordinator._payForRange: startTimestamp too far in the future"
         );
 
-        // Require rangePayment is for whitelisted strategies
+        // Require rangePayment is for whitelisted strategy or beaconChainETHStrategy
         for (uint256 j = 0; j < rangePayment.strategiesAndMultlipliers.length; j++) {
             IStrategy strategy = rangePayment.strategiesAndMultlipliers[j].strategy;
             require(
-                strategyManager.strategyIsWhitelistedForDeposit(strategy),
-                "PaymentCoordinator.payForRange: strategy not whitelisted for deposit"
+                strategyManager.strategyIsWhitelistedForDeposit(strategy) ||
+                    strategy == delegationManager.beaconChainETHStrategy(),
+                "PaymentCoordinator._payForRange: invalid strategy considered"
             );
         }
 
         // Set hash of rangePayment in mapping
-        bytes32 rangePaymentHash = keccak256(abi.encode(msg.sender, paymentNonce, rangePayment));
-        paymentNonce++;
         isRangePaymentHash[msg.sender][rangePaymentHash] = true;
+        paymentNonce++;
 
         rangePayment.token.safeTransferFrom(msg.sender, address(this), rangePayment.amount);
         return rangePaymentHash;
     }
 
     /**
-     * @notice Claim payments for the given claim
-     * @param claim The claims to be processed
+     * @notice Process a payment claim.
+     * @param claim PaymentMerkleClaim struct containing the claim details for proof verification
+     * and payment transfer.
+     * @dev only callable by claimerFor[claim.earner]
+     */
+    function _processClaim(PaymentMerkleClaim calldata claim) internal onlyWhenNotPaused(PAUSED_CLAIM_PAYMENTS) {
+        DistributionRoot memory root = distributionRoots[claim.rootIndex];
+        require(
+            block.timestamp > root.activatedAt + activationDelay,
+            "PaymentCoordinator._processClaim: root not activated yet"
+        );
+
+        // If claimerFor earner is not set, claimer is by default the earner. Else set to claimerFor
+        address claimer = claimerFor[claim.earner] == address(0) ? claim.earner : claimerFor[claim.earner];
```

Comment:
this will do 2 sloads if im not wrong
```solidity
address claimer = claimerFor[claim.earner];
if (claimer == address(0)) {
      claimer = claim.earner;
}
```

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/blob/8bcbaf830523cf49949011d507b4750cd221ad67/src/contracts/core/PaymentCoordinator.sol#L236